src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -865,13 +870,9 @@ private async Task<ConnectBlocksResult> ConnectBlockAsync(ChainedHeaderBlock blo
             {
                 var badPeers = new List<int>();
 
-                // Ban the peers only in case block is invalid and not temporary rejected.
-                if (validationContext.RejectUntil == null)
```

Comment:
This will remove the header and its decedents from the tree, are you sure you want to do that if the ban temporary?

Reply:
Yes. That's basically what will retrigger the whole download sequence from header to FV block

Reply:
I believe it will only indirectly trigger the download sequence (using the get header timer) this is what you mean?

Reply:
Yes